lxd/cluster/notify.go

```diff
@@ -43,11 +48,12 @@ func NewNotifier(state *state.State, networkCert *shared.CertInfo, serverCert *s
 
 	var err error
 	var members []db.NodeInfo
-	var offlineThreshold time.Duration
 	err = state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-		offlineThreshold, err = tx.GetNodeOfflineThreshold(ctx)
-		if err != nil {
-			return err
+		if state.GlobalConfig == nil {
+			offlineThreshold, err = tx.GetNodeOfflineThreshold(ctx)
```

Comment:
Could you add a comment to the effect of the commit message here explaining why we need it.

Reply:
which invocation of NewNotifier is called when there's no global DB btw?

Reply:
[`NotifyUpgradeCompleted`](https://github.com/canonical/lxd/blob/main/lxd/cluster/upgrade.go#L24), called [in the daemon](https://github.com/canonical/lxd/blob/main/lxd/daemon.go#L1390) before `patchesApply(d, patchPreLoadClusterConfig`. I didn't want to side-quest too far into why `NotifyUpgradeCompleted` is happening before we apply patches. Too late I guess:

I think the cluster notification has to happen before patches are applied so that we don't end up deadlocking without quorum:
- Old server starts
- New server starts, fails to open DB and waits for old server to send upgrade notification
- Old server is upgraded, restarts, starts global db and sends upgrade notification
- New server receives upgrade notification and retries opening the DB
- DB has reached quorum and both servers can run patches
If the old server tries to run patches or otherwise access the DB before sending a notification, the DB will only be started on that member and will not have quorum

Reply:
I also considered using the default offline threshold instead of going back to the DB, but I think this is probably more correct.